nautobot/utilities/forms/fields.py

```diff
@@ -788,3 +768,25 @@ def _check_values(self, values):
         if null:
             result += [self.null_value]
         return result
+
+
+class TagFilterField(DynamicModelMultipleChoiceField):
+    """
+    A filter field for the tags of a model. Only the tags used by a model are displayed.
+
+    :param model: The model of the filter
+    """
+
+    def __init__(self, model, *args, query_params=None, queryset=None, **kwargs):
+        queryset = queryset or model.tags.all()
```

Comment:
I'm not sure if there's a use case where this would be encountered but if `queryset` is an empty queryset (`Tag.objects.filter(name="nonexistent")`), it would fall back to `model.tags.all()`. Is that the behavior we want or should it be `if queryset is None:`?

Reply:
`if None` would be better. Good catch!

Reply:
also saves us evaluating the queryset at init time.